src/app/data-client.ts

```diff
@@ -67,6 +69,65 @@ export class DataClient {
     return dataArray;
   }
 
+  async binaryDataByFilter(filter?: pb.Filter) {
+    const { service } = this;
+
+    let last = '';
+    const dataArray: pb.BinaryData.AsObject[] = [];
+
+    for (;;) {
+      const dataReq = new pb.DataRequest();
+      if (filter) {
+        dataReq.setFilter(filter);
+      } else {
+        dataReq.setFilter(new pb.Filter());
+      }
+      dataReq.setLimit(100);
+      dataReq.setLast(last);
+
+      const req = new pb.BinaryDataByFilterRequest();
+      req.setDataRequest(dataReq);
+      req.setCountOnly(false);
+
+      // eslint-disable-next-line no-await-in-loop
+      const response = await promisify<
+        pb.BinaryDataByFilterRequest,
+        pb.BinaryDataByFilterResponse
+      >(service.binaryDataByFilter.bind(service), req);
+      const dataList = response.getDataList();
+      if (!dataList || dataList.length === 0) {
+        break;
+      }
+      dataArray.push(...dataList.map((data) => data.toObject()));
+      last = response.getLast();
+    }
+
+    return dataArray;
+  }
+
+  async binaryDataByIds(ids: BinaryID[]) {
+    const { service } = this;
+
+    const binaryIds: pb.BinaryID[] = [];
+    for (const id of ids) {
+      const binaryId = new pb.BinaryID();
+      binaryId.setFileId(id.fileId);
+      binaryId.setOrganizationId(id.organizationId);
+      binaryId.setLocationId(id.locationId);
+      binaryIds.push(binaryId);
+    }
```

Comment:
Similar to how the tabular data response was mapped into the data list, the `binaryIds` list can be created by mapping over the list of `ids`:

```suggestion
    const binaryIds: pb.BinaryID[] = ids.map(({ fileId, organizationId, locationId }) => {
      const binaryId = new pb.BinaryID();
      binaryId.setFileId(fileId);
      binaryId.setOrganizationId(organizationId);
      binaryId.setLocationId(locationId);
      return binaryId;
    });
```